onnxruntime/core/providers/qnn/builder/opbuilder/resize_op_builder.cc

```diff
@@ -38,51 +42,227 @@ class ResizeOpBuilder : public BaseOpBuilder {
                                       const NodeUnit& node_unit,
                                       const logging::Logger& logger,
                                       bool is_quantized_model) const {
-  ORT_UNUSED_PARAMETER(logger);
-  ORT_UNUSED_PARAMETER(is_quantized_model);
+  if (node_unit.Domain() == kMSInternalNHWCDomain) {
+    return AddToModelBuilder(qnn_model_wrapper, node_unit, logger, is_quantized_model, true);
+  }
+
+  // QNN doesn't support anti-aliasing (added in opset 18)
+  if (node_unit.SinceVersion() >= 18) {
+    NodeAttrHelper node_helper(node_unit);
+    const bool antialias = GetOnnxAttr(node_helper, onnx_antialias_attr) != 0;
+    ORT_RETURN_IF(antialias, "QNN EP: Resize doesn't support anti-aliasing.");
+  }
+
+  // The QNN Resize op does not currently work with the QNN cpu backend, but works with the HTP backend. Therefore, we
+  // currently use QNN's Resize op for quantized models and either ResizeBilinear or ResizeNearestNeighbor for
+  // non-quantized models. This requires separate validation for quantized models.
+  // TODO: Use only Resize once QNN's Resize op works in the QNN cpu backend.
+  return is_quantized_model ? ValidateQDQOp(qnn_model_wrapper, node_unit) : ValidateOp(qnn_model_wrapper, node_unit);
+}
+
+Status ResizeOpBuilder::ValidateOp(QnnModelWrapper& qnn_model_wrapper, const NodeUnit& node_unit) const {
   NodeAttrHelper node_helper(node_unit);
-  const std::string& resize_mode = node_helper.Get("mode", "nearest");
-  ORT_RETURN_IF("cubic" == resize_mode, "Resize donesn't support cubic mode!");
-
-  const std::string& coordinate_mode = node_helper.Get("coordinate_transformation_mode", "half_pixel");
-  if ("pytorch_half_pixel" == coordinate_mode ||
-      "tf_crop_and_resize" == coordinate_mode ||
-      "tf_half_pixel_for_nn" == coordinate_mode) {
-    std::string msg = coordinate_mode + " mode not supported, QNN Resize only support align_corners and half_pixel";
+  const std::string resize_mode = GetOnnxAttr(node_helper, onnx_mode_attr);
+  ORT_RETURN_IF("cubic" == resize_mode, "Resize doesn't support cubic mode!");
+
+  const std::string coordinate_mode = GetOnnxAttr(node_helper, onnx_coord_transf_mode_attr);
+  if ((coordinate_mode != "half_pixel") && (coordinate_mode != "align_corners")) {
+    std::string msg = coordinate_mode + " mode not supported, QNN Resize only supports align_corners and half_pixel";
     return ORT_MAKE_STATUS(ONNXRUNTIME, FAIL, msg);
   }
 
-  if ("nearest" == resize_mode) {
-    const std::string& nearest_mode = node_helper.Get("nearest_mode", "round_prefer_floor");
-    ORT_RETURN_IF_NOT("floor" == nearest_mode, "QNN Resize only support nearest_mode: floor!");
+  if (resize_mode == "nearest") {
+    const std::string& nearest_mode = GetOnnxAttr(node_helper, onnx_nearest_mode_attr);
+    ORT_RETURN_IF_NOT("round_prefer_floor" == nearest_mode, "QNN Resize only supports nearest_mode: round_prefer_floor!");
```

Comment:
Unit tests revealed that QNN's ResizeNearestNeighbor op actually uses `round_prefer_floor`.